networkit/cpp/graph/GraphBuilder.cpp

```diff
@@ -101,60 +180,64 @@ void GraphBuilder::swapNeighborhood(node u, std::vector<node> &neighbours,
 }
 void GraphBuilder::setOutWeight(node u, node v, edgeweight ew) {
     assert(isWeighted());
-    index vi = indexInOutEdgeArray(u, v);
+    index vi = indexInOutEdgeArrayPerThread(u, v);
     if (vi != none) {
-        outEdgeWeights[u][vi] = ew;
+        outEdgeWeightsPerThread[omp_get_thread_num()][u % omp_get_max_threads()][vi] = ew;
+        if (!directed && autoCompleteEdges) { // need to adjust both half edges
+            index ui = indexInOutEdgeArrayPerThread(v, u);
+            outEdgeWeightsPerThread[omp_get_thread_num()][v % omp_get_max_threads()][ui] = ew;
+        }
     } else {
-        addHalfOutEdge(u, v, ew);
+        addHalfEdge(u, v, ew);
     }
 }
 
 void GraphBuilder::setInWeight(node u, node v, edgeweight ew) {
     assert(isWeighted());
     assert(isDirected());
-    index vi = indexInInEdgeArray(u, v);
+    index vi = indexInInEdgeArrayPerThread(u, v);
     if (vi != none) {
-        inEdgeWeights[u][vi] = ew;
+        inEdgeWeightsPerThread[omp_get_thread_num()][u % omp_get_max_threads()][vi] = ew;
     } else {
-        addHalfInEdge(u, v, ew);
+        addHalfEdge(u, v, ew);
     }
 }
 
 void GraphBuilder::increaseOutWeight(node u, node v, edgeweight ew) {
     assert(isWeighted());
-    index vi = indexInOutEdgeArray(u, v);
+    index vi = indexInOutEdgeArrayPerThread(u, v);
     if (vi != none) {
-        outEdgeWeights[u][vi] += ew;
+        outEdgeWeightsPerThread[omp_get_thread_num()][u % omp_get_max_threads()][vi] += ew;
+        if (!directed && autoCompleteEdges && u != v) { // need to adjust both half edges
+            index ui = indexInOutEdgeArrayPerThread(v, u);
+            outEdgeWeightsPerThread[omp_get_thread_num()][v % omp_get_max_threads()][ui] += ew;
+        }
     } else {
-        addHalfOutEdge(u, v, ew);
+        addHalfEdge(u, v, ew);
     }
 }
 
 void GraphBuilder::increaseInWeight(node u, node v, edgeweight ew) {
     assert(isWeighted());
     assert(isDirected());
-    index vi = indexInInEdgeArray(u, v);
+    index vi = indexInInEdgeArrayPerThread(u, v);
     if (vi != none) {
-        inEdgeWeights[u][vi] += ew;
+        inEdgeWeightsPerThread[omp_get_thread_num()][u % omp_get_max_threads()][vi] += ew;
     } else {
-        addHalfInEdge(u, v, ew);
+        addHalfEdge(u, v, ew);
     }
 }
 
-Graph GraphBuilder::completeGraph(bool parallel) {
```

Comment:
The deprecated implementation is missing. Look at other deprecated functions. The implementation includes a warning and usually a subsequent call to the new implementation.

Example:
```
void MyClass:myDeprecatedFunction(...) {
    WARN("DEPRECATED: Use function ... instead.");
    newFunction(...);
}
```